### ISSUE TITLE
Documentation Refactor

### ISSUE THREAD

**Issue by fvisin** [Theano/Theano#4141]
The scope of this ticket is to go through the documentation of theano and fix or report mistakes and not easily comprehensible parts of it.

**Comment by JesseLivezey** [comment]
I personally think some of the ```scan``` documentation could be reorganized to make it easier to use. 

There is this page:
http://deeplearning.net/software/theano/library/scan.html
Two main ideas:
 - There are a bunch of examples before the ```scan``` function and signature are really described. I would move the ```reference``` part of the page before the ```Guide``` part of the page. Also ```reference``` could be capitalized if changes are made.
 - In the ```reference``` part, ```map```, ```reduce```, ```foldl(r)```, are all described before ```scan```, but they all reference ```scan``` in their documentation. I think it would make more sense to have ```scan``` at the top of the page, then the other functions, then the ```Guide``` below them.

Then there is this page:
http://deeplearning.net/software/theano/tutorial/loop.html
This page seems somewhat redundant with the ```Guide``` page in that it would probably be used by someone who is trying to figure out how to use ```scan``` versus someone who just needs to remember the order or meaning of arguments and just wants the ```reference```.

**Comment by gokul-uf** [comment]
@fvisin is this ticket open for all or is it assigned to someone? I remember seeing in some other issue that CCWs are usually coassigned to someone. If not, let me know how I could help!

**Comment by fvisin** [comment]
@JesseLivezey Thank you for your suggestions! This ticket was initially meant to address the high level organization of the documentation, but I guess we will also get into the details of some of the pages. Your comments are very helpful, should we not incorporate them in this PR I will open another more focalized issue. Please feel welcome to provide other ideas.

@gokul-uf You are right, issues tagged as `CCW` are usually assigned internally. This one is currently being addressed by #4211 (and in general documentation issues will be usually addressed internally because they require a lot of brainstorming and interaction that is much easier in person)

**Comment by mgermain** [comment]
Just a note about something which is not documented.

In `.theanorc` the way I would put this flag `config.dnn.conv.algo_fwd` is:
```
[dnn]
conv.algo_fwd=small
```
But .theanorc wants max one nesting in sections flags so the right way to do it is:
```
[dnn.conv]
algo_fwd=small
```

**Comment by mgermain** [comment]
Just noticed that the documentation about cuDNN is out of sync in the old and new backend and is out of date.

NEW: doc/library/sandbox/gpuarray/dnn.py
OLD: doc/library/sandbox/cuda/dnn.py

We should probably just add a note about the upcoming deprecation of the old backend and update the new backend doc with all the supported algorithms for convolutions.

**Comment by fvisin** [comment]
Thanks, I added it to the todo list. This is going to be a neverending ticket I fear!

**Comment by mgermain** [comment]
Héhé, If there is sticky issues in git this one should but one :P

**Comment by fvisin** [comment]
Ahahahah definitely! :)

**Comment by etdej** [comment]
This page is not visualized correctly: http://deeplearning.net/software/theano/sandbox/broadcasting.html

**Comment by nouiz** [comment]
Good idea to fix this, but this is in a doc sandbox directory. I think we
should make a good broadcast tutorial and not just clear this one op.

I made a separate issue: https://github.com/Theano/Theano/issues/4519

**Comment by fvisin** [comment]
Agreed, thanks for the comment.
